src/site/_transforms/service-worker-partials/index.js

```diff
@@ -38,16 +49,72 @@ const getPartial = (content) => {
     title: $('title').text(),
     rss: $('link[type="application/atom+xml"]').attr('href'),
     offline: Boolean($('meta[name="offline"]').attr('content')) || false,
+    ...revision,
   };
   return partial;
 };
 
+const writeTemplate = async (to, raw) => {
+  const manifest = [];
+
+  // In prod, fetch manifest assets. We don't bother in dev as we're not sure
+  // of the execution order.
+  if (isProd) {
+    const toplevelManifest = await getManifest({
+      // JS files that include hashes don't need their own revision fields.
+      globDirectory: 'dist',
+      globPatterns: [
+        // We don't include jpg files, as they're used for authors and hero
+        // images, which are part of articles, and not the top-level site.
+        'images/**/*.{png,svg}',
+        '*.css',
+        '*.js',
+        '*.partial',
+      ],
+      globIgnores: [
+        // This removes large shared PNG files that are used only for articles.
+        'images/{shared}/**',
+      ],
+    });
+    if (toplevelManifest.warnings.length) {
+      throw new Error(`toplevel manifest: ${toplevelManifest.warnings}`);
+    }
+    manifest.push(...toplevelManifest.manifestEntries);
+
+    // We need this manifest to be separate as we pretend it's rooted at the
+    // top-level, even though it comes from "dist/en".
+    const contentManifest = await getManifest({
+      globDirectory: 'dist/en',
+      globPatterns: ['offline/index.json'],
```

Comment:
is `offline/index.json` guaranteed to exist at this point?

Reply:
No, I think it's probably unknown since we're relying on Eleventy's output order.